src/execution/execute-afm.ts

```diff
@@ -1,137 +1,251 @@
-                return this.getOnePage(pollingUri, offset)
-                    .then((executionResultWrapper: Execution.IExecutionResultWrapper) => {
-                        const responses: Execution.IExecutionResponses = {
-                            executionResponse: executionResponseWrapper.executionResponse,
-                            executionResult: executionResultWrapper ? executionResultWrapper.executionResult : null
-                        };
-                        return responses;
+    public executeAfm(projectId: string, execution: AFM.IExecution)
+        : Promise<Execution.IExecutionResponses> {
+        validateNumOfDimensions(get(execution, 'execution.resultSpec.dimensions').length);
+        return this.getExecutionResponse(projectId, execution)
+            .then((executionResponse: Execution.IExecutionResponse) => {
+                return this.getWholeExecutionResult(executionResponse.links.executionResult)
+                    .then((executionResult: Execution.IExecutionResult | null) => {
+                        return { executionResponse, executionResult };
                     });
             });
     }
 
-    private fetchExecutionResult(pollingUri: string, offset: number[]) {
-        const [uriPart, queryPart] = pollingUri.split(/\?(.+)/);
-        const query = {
-            ...qs.parse(queryPart),
-            limit: getLimit(offset).join(','),
-            offset: offset.join(',')
-        };
-        const finalPollingUri = uriPart + qs.stringify(query, { addQueryPrefix: true });
-        return this.xhr.ajax(finalPollingUri, { method: 'GET' }).then((r) => {
-            if (r.response.status === 204) {
-                return null;
-            }
-            return r.getData();
-        });
+    /**
+     * Get Response from Execution
+     *
+     * @method getExecutionResponse
+     * @param {string} projectId - GD project identifier
+     * @param {AFM.IExecution} execution - See https://github.com/gooddata/gooddata-typings/blob/master/index.ts#L4
+     *
+     * @returns {Promise<Execution.IExecutionResponse>} Promise with `executionResponse`
+     *  See https://github.com/gooddata/gooddata-typings/blob/master/src/Execution.ts#L69
+     */
+    private getExecutionResponse(projectId: string, execution: AFM.IExecution)
+        : Promise<Execution.IExecutionResponse> {
+        validateNumOfDimensions(get(execution, 'execution.resultSpec.dimensions').length);
+        return this.xhr.post(`/gdc/app/projects/${projectId}/executeAfm`, { body: JSON.stringify(execution) })
+            .then(apiResponse => apiResponse.getData())
+            .then(unwrapExecutionResponse);
     }
 
-    private getOnePage(
-        pollingUri: string, offset: number[], prevResultWrapper?: Execution.IExecutionResultWrapper
-    ): any {
-        return this.fetchExecutionResult(pollingUri, offset)
+    /**
+     * Get whole (all pages) Result from Execution
+     *
+     * @method getWholeExecutionResult
```

Comment:
I think `getFullExecutionResult` sounds better, or inverse naming and use `getPartialExecutionResult` 

what do you think? :)

Reply:
I would prefer something like `getAllResultPages`

Reply:
getAllResultPages return Promise<Execution.IExecutionResult> its quite confusing function with this name I would expect object like ResultPage which is not true...

Reply:
@martinmosko @zbycz please give me the final decision

Reply:
renamed